### PR TITLE
update contributing guide and ui-tests scripts

### PR DESCRIPTION
**Pull request by mordamax**
- Updated ./docs/CONTRIBUTION.md to clarify how to use command bot and update ui-tests locally or in PR
- Moved update-ui-tests.sh to a root of project to run them along for substrate and polkadot together

### CODE REVIEW

**Comment by mordamax** [comment]
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Comment by mordamax** [comment]
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Comment by mordamax** [comment]
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Comment by mordamax** [comment]
bot clean
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

File: substrate/.maintain/update-rust-stable.sh [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-
-rustup install $RUST_VERSION
-rustup component add rust-src --toolchain $RUST_VERSION
-
-# Ensure we run the ui tests
-export RUN_UI_TESTS=1
-# We don't need any wasm files for ui tests
-export SKIP_WASM_BUILD=1
-# Let trybuild overwrite the .stderr files
-export TRYBUILD=overwrite
```
Comment (altaua) [inline substrate/.maintain/update-rust-stable.sh:34]: These variables need to be added to the new scripts.

Comment (mordamax) [reply substrate/.maintain/update-rust-stable.sh:34]: thx lost them with moving b&f

**Comment by mordamax** [comment]
bot clean
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Comment by mordamax** [comment]
bot update-ui  stable -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests --rust_version=1.70

**Comment by mordamax** [comment]
bot update-ui latest -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests --rust_version=1.70

**Comment by mordamax** [comment]
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Review by bkchr — APPROVED** [review]

**Comment by mordamax** [comment]
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Comment by mordamax** [comment]
@bkchr `cargo test -p orchestra ui` failed with `
error: package `orchestra` cannot be tested because it requires dev-dependencies and is not a member of the workspace`
do you know how to fix? or just exclude for now?

**Review by sergejparity — APPROVED** [review]

**Comment by mordamax** [comment]
bot clean
bot update-ui latest -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests --rust_version=1.70

**Comment by mordamax** [comment]
bot clean
bot update-ui latest -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests --rust_version=1.70

**Comment by bkchr** [comment]
> @bkchr `cargo test -p orchestra ui` failed with `error: package "orchestra" cannot be tested because it requires dev-dependencies and is not a member of the workspace` https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3766421 do you know how to fix? or just exclude for now?

Yeah the crate doesn't exist anymore in our repo, so you can just drop the line.

**Comment by mordamax** [comment]
bot clean
bot update-ui -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests

**Review by ggwpez — APPROVED** [review]

File: scripts/update-ui-tests.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Script for updating the UI tests for a new rust stable version.
+# Exit on error
+set -e
+
+# by default current rust stable will be used
+RUSTUP_RUN=""
+# check if we have a parameter
+# ./substrate/.maintain/update-rust-stable.sh 1.70
```
Comment (altaua) [inline scripts/update-ui-tests.sh:9]:
```suggestion
# ./scripts/update-ui-tests.sh 1.70
```

File: scripts/update-ui-tests.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  rustup install $RUST_VERSION
+  rustup component add rust-src --toolchain $RUST_VERSION
+
+
+  if ! command -v rustup &> /dev/null
+  then
+    echo "rustup needs to be installed"
+    exit
+  fi
```
Comment (altaua) [inline scripts/update-ui-tests.sh:24]:
```suggestion
  if ! command -v rustup &> /dev/null
  then
    echo "rustup needs to be installed"
    exit
  fi
  
  rustup install $RUST_VERSION
  rustup component add rust-src --toolchain $RUST_VERSION
```

**Review by altaua — APPROVED** [review]

File: scripts/update-ui-tests.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+# Script for updating the UI tests for a new rust stable version.
+# Exit on error
+set -e
+
+# by default current rust stable will be used
+RUSTUP_RUN=""
+# check if we have a parameter
+# ./substrate/.maintain/update-rust-stable.sh 1.70
+if [ "$#" -e 1 ]; then
```
Comment (altaua) [inline scripts/update-ui-tests.sh:10]: This check isn't quite sufficient: command-bot calls this script with an "" argument if no rust version was specified, in which case $# is 1 but $1 is "".

Comment (altaua) [reply scripts/update-ui-tests.sh:10]: Oh and also there's a typo here, the equality operator is `-eq`, not `-e` :)

Comment (mordamax) [reply scripts/update-ui-tests.sh]: yea, actually -z "$1" is  what i had to use :) fixed

**Comment by mordamax** [comment]
bot update-ui latest -v PIPELINE_SCRIPTS_REF=mak-cleanup-update-ui-tests --rust_version=1.70

**Review by altaua — APPROVED** [review]

**Pull request merged**